simplipy/api.py

```diff
@@ -303,37 +303,40 @@ async def request(  # pylint: disable=too-many-branches
             try:
                 resp.raise_for_status()
             except ClientError as err:
-                if data.get("error") == "mfa_required":
-                    return data
-
-                if data.get("type") == "NoRemoteManagement":
-                    raise EndpointUnavailable(
-                        f"Endpoint unavailable in plan: {endpoint}"
-                    ) from None
-
-                if "401" in str(err):
-                    if self._actively_refreshing:
-                        raise InvalidCredentialsError(
-                            "Repeated 401s despite refreshing access token"
-                        ) from None
-                    if self._refresh_token:
-                        _LOGGER.info("401 detected; attempting refresh token")
-                        self._access_token_expire = datetime.now()
-                        return await self.request(method, endpoint, **kwargs)
-                    raise InvalidCredentialsError("Invalid username/password") from None
-
-                if "403" in str(err):
-                    raise InvalidCredentialsError("Invalid username/password") from None
-
-                raise RequestError(
-                    f"There was an error while requesting /{endpoint}: {err}"
-                ) from None
+                return await self.handle_error(data, method, endpoint, err, kwargs)
```

Comment:
I don't think this is the right approach – it's unclear what exactly `handle_error` is responsible for, especially with such a large argument signature.

I'd recommend seeing if this can be broken up into multiple, smaller methods, each with a clear understanding of what its args belong to it.